src/main/java/appeng/fluids/storage/FluidCellInventory.java

```diff
@@ -0,0 +1,285 @@
+			{
+				final IAEFluidStack r = input.copy();
+				r.setStackSize( r.getStackSize() - remainingItemSlots );
+				if( mode == Actionable.MODULATE )
+				{
+					l.setStackSize( l.getStackSize() + remainingItemSlots );
+					this.updateItemCount( remainingItemSlots );
+					this.saveChanges();
+				}
+				return r;
+			}
+			else
+			{
+				if( mode == Actionable.MODULATE )
+				{
+					l.setStackSize( l.getStackSize() + input.getStackSize() );
+					this.updateItemCount( input.getStackSize() );
+					this.saveChanges();
+				}
+				return null;
+			}
+		}
+
+		if( this.canHoldNewItem() ) // room for new type, and for at least one item!
+		{
+			final int remainingItemCount = (int) this.getRemainingItemCount() - this.getBytesPerType() * 8;
+			if( remainingItemCount > 0 )
+			{
+				if( input.getStackSize() > remainingItemCount )
+				{
+					final FluidStack toReturn = sharedFluidStack.copy();
+					toReturn.amount = sharedFluidStack.amount - remainingItemCount;
+					if( mode == Actionable.MODULATE )
+					{
+						final FluidStack toWrite = sharedFluidStack.copy();
+						toWrite.amount = remainingItemCount;
+
+						this.cellItems.add( AEFluidStack.fromFluidStack( toWrite ) );
+						this.updateItemCount( toWrite.amount );
+
+						this.saveChanges();
+					}
+					return AEFluidStack.fromFluidStack( toReturn );
+				}
+
+				if( mode == Actionable.MODULATE )
+				{
+					this.updateItemCount( input.getStackSize() );
+					this.cellItems.add( input );
+					this.saveChanges();
+				}
+
+				return null;
+			}
+		}
+
+		return input;
+	}
+
+	@Override
+	public IAEFluidStack extractItems( final IAEFluidStack request, final Actionable mode, final IActionSource src )
+	{
+		if( request == null )
+		{
+			return null;
+		}
+
+		final long size = Math.min( Integer.MAX_VALUE, request.getStackSize() );
+
+		IAEFluidStack Results = null;
+
+		final IAEFluidStack l = this.getCellItems().findPrecise( request );
+		if( l != null )
+		{
+			Results = l.copy();
+
+			if( l.getStackSize() <= size )
+			{
+				Results.setStackSize( l.getStackSize() );
+				if( mode == Actionable.MODULATE )
+				{
+					this.updateItemCount( -l.getStackSize() );
+					l.setStackSize( 0 );
+					this.saveChanges();
+				}
+			}
+			else
+			{
+				Results.setStackSize( size );
+				if( mode == Actionable.MODULATE )
+				{
+					l.setStackSize( l.getStackSize() - size );
+					this.updateItemCount( -size );
+					this.saveChanges();
+				}
+			}
+		}
+
+		return Results;
+	}
+
+	@Override
+	public IStorageChannel getChannel()
+	{
+		return AEApi.instance().storage().getStorageChannel( IFluidStorageChannel.class );
+	}
+
+	protected void loadCellItem( NBTTagCompound compoundTag, int stackSize )
+	{
+
+		// Now load the fluid stack
+		final FluidStack t;
+		try
+		{
+			t = FluidStack.loadFluidStackFromNBT( compoundTag );
+			if( t == null )
+			{
+				AELog.warn( "Removing item " + compoundTag + " from storage cell because the associated item type couldn't be found." );
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "Removing item " 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1192)